pkg/apis/maistra/v2/servicemeshcontrolplane_types.go

```diff
@@ -202,3 +202,12 @@ func (s ControlPlaneSpec) IsClusterScoped() (bool, error) {
 	}
 	return controlPlaneMode == ControlPlaneModeValueClusterScoped, nil
 }
+
+func (s ControlPlaneSpec) IsPilotSecretNameEnabled() bool {
+	return s.Security.CertificateAuthority.CertManager != nil && s.Security.CertificateAuthority.CertManager.PilotCertSecretName != ""
```

Comment:
`Security` and `CertificateAuthority` are pointers, so you need to check if they're `nil`.